src/components/model/ModelForm.svelte

```diff
@@ -0,0 +1,167 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { createEventDispatcher, onDestroy } from 'svelte';
+  import { resetModelStores } from '../../stores/model';
+  import type { User, UserId } from '../../types/app';
+  import { permissionHandler } from '../../utilities/permissionHandler';
+  import { featurePermissions } from '../../utilities/permissions';
+  import { getShortISOForDate } from '../../utilities/time';
+  import { tooltip } from '../../utilities/tooltip';
+  import Input from '../form/Input.svelte';
+
+  export let initialModelDescription: string = '';
+  export let initialModelName: string = '';
+  export let initialModelOwner: UserId | undefined;
+  export let initialModelVersion: string | undefined;
+  export let modelId: number | undefined;
+  export let createdAt: string | undefined;
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher<{
+    createPlan: number;
+    deleteModel: void;
+    hasModelChanged: {
+      description: string;
+      name: string;
+      owner: UserId;
+      version: string;
+    };
+  }>();
+
+  const updateModelPermissionError: string = 'You do not have permission to update this model';
+  const deleteModelPermissionError: string = 'You do not have permission to delete this model';
+  const createPlanPermissionError: string = 'You do not have permission to create a plan';
+
+  let hasUpdateModelPermission: boolean = false;
+  let hasDeleteModelPermission: boolean = false;
+  let hasCreatePlanPermission: boolean = false;
+  let name: string = '';
+  let owner: UserId | null = null;
+  let version: string = '';
+  let description: string = '';
+
+  $: description = initialModelDescription;
+  $: name = initialModelName;
+  $: owner = initialModelOwner ?? null;
+  $: version = initialModelVersion ?? '';
+
+  $: if (user) {
+    hasUpdateModelPermission = featurePermissions.model.canUpdate(user);
+    hasDeleteModelPermission = featurePermissions.model.canDelete(user);
+    hasCreatePlanPermission = featurePermissions.plan.canCreate(user);
+  }
+
+  $: dispatch('hasModelChanged', {
+    description,
+    name,
+    owner,
+    version,
+  });
+
+  function onCreatePlanWithModel() {
+    if (modelId !== undefined) {
+      dispatch('createPlan', modelId);
+    }
+  }
+
+  function onDeleteModel() {
+    dispatch('deleteModel');
+  }
+
+  onDestroy(() => {
+    resetModelStores();
+  });
+</script>
+
+<div class="model-form-container">
+  <Input layout="inline">
+    <label for="name">Model name</label>
+    <input
+      class="st-input w-100"
+      name="name"
```

Comment:
Consider adding `autocomplete="off"` to avoid native autocompleting on this input
<img width="471" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/49248c40-649b-45b2-966c-209055609e36">